### PR TITLE
Add Lighthouse into projen template

### PR DESCRIPTION
**Pull request by bdoof**

### CODE REVIEW

File: src/common/github/pull-request-test-workflow.ts [hunk 1]
```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
...
       typecheck: job('npm run typecheck'),
       test: job('npm run test'),
     })
+
+    if (options.lighthouse) {
+      workflow.addJobs({
+        lighthouse: customJob([
+          {uses: 'actions/checkout@v3'},
+          {uses: 'actions/setup-node@v3', with: {'node-version': 16}},
+          {name: 'Install dependencies', run: 'npm instapp'},
```
Comment (quesabe) [inline src/common/github/pull-request-test-workflow.ts:81]: `npm install` or maybe `npm ci`

File: src/common/github/pull-request-test-workflow.ts [hunk 2]
```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
       steps: [{uses: 'ottofeller/github-actions/npm-run@main', with: {'node-version': 16, command, directory}}],
     })
 
+    const customJob = (steps: JobStep[]): github.workflows.Job => ({
```
Comment (quesabe) [inline src/common/github/pull-request-test-workflow.ts:64]: There is a `job` function in `src/common/github/job.ts`. We can utilise that one here adding a second optional argument.
```typescript
export const job = (steps: Array<JobStep>, runsOn = ['ubuntu-latest']): Job => ({
  runsOn,
  permissions: {contents: projen.github.workflows.JobPermission.READ},
  steps,
})
```

However I'm not sure we pass `runsOn` anywhere - moreover seems like it's not even possible with current setup. So you'd be able to use the function as is.

File: src/common/github/pull-request-test-workflow.ts [hunk 3]
```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
...
+      steps,
+    })
+
     workflow.addJobs({
       lint: job('npm run lint'),
       typecheck: job('npm run typecheck'),
       test: job('npm run test'),
     })
+
+    if (options.lighthouse) {
```
Comment (quesabe) [inline src/common/github/pull-request-test-workflow.ts:76]: This works but seems a little weird. The workflow is used by all projects and adding an option that is needed only in a single one looks not SOLID. Maybe we could add some general purpose extension mechanism (like a customJobs array option) and define the details of these custom jobs in the nextjs project.

However I would not object if @gvidon is ok with it. If we have plans on adding any other frontend template the lighthouse audit step can definitely be consider a part of the workflow template.

**Review by quesabe — CHANGES_REQUESTED** [review]

File: src/common/github/pull-request-test-workflow.ts (hunk 3 above)
Comment (gvidon) [reply src/common/github/pull-request-test-workflow.ts:85]: @quesabe agree. I would probably suggest adding new workflow just for the nextjs projects, by extending this one with lighthouse, and also playwright in the future.

File: src/common/github/pull-request-test-workflow.ts [hunk 4]
```diff
@@ -53,11 +61,36 @@ export class PullRequestTest extends Component {
...
+          {uses: 'actions/setup-node@v3', with: {'node-version': 16}},
+          {name: 'Install dependencies', run: 'npm instapp'},
+          {name: 'Copy environment variables', run: 'cp .env.development .env.local'},
+          {name: 'Build Next.js application', run: 'npm run build'},
+          {name: 'Run Lighthouse audit', run: 'npm run lighthouse'},
+          {
+            name: 'Save Lighthouse report as an artifact',
+            uses: 'actions/upload-artifact@v3',
+            if: 'always()',
+            with: {name: 'lighthouse-report', path: '.lighthouseci/'},
```
Comment (gvidon) [inline src/common/github/pull-request-test-workflow.ts:98]: @bdoof should this path be customisable?

File: src/nextjs/index.ts [hunk 5]
```diff
@@ -37,6 +37,13 @@ export interface OttofellerNextjsProjectOptions
    * @default true
    */
   readonly isUiConfigEnabled?: boolean
+
+  /**
+   * Setup Lighthouse audit script & job.
```
Comment (gvidon) [inline src/nextjs/index.ts:42]: @bdoof nit pick
```suggestion
   * Setup Lighthouse audit script & GitHub job.
```

**Review by gvidon — CHANGES_REQUESTED** [review]

File: src/common/github/pull-request-test-workflow.ts (hunk 3 above)
Comment (gvidon) [reply src/common/github/pull-request-test-workflow.ts:85]: @bdoof yeah, I think new PR is a good idea.

**Review by quesabe — APPROVED** [review]

File: src/common/github/pull-request-test-workflow.ts [hunk 6]
```diff
@@ -27,6 +28,13 @@ export interface PullRequestTestOptions {
    * @default ['main']
    */
   readonly triggerOnPushToBranches?: Array<string>
+
+  /**
+   * Setup Lighthouse audit job.
+   *
+   * @default true
+   */
+  readonly lighthouse?: boolean
```
Comment (gvidon) [inline src/common/github/pull-request-test-workflow.ts:37]: @bdoof I would rename it to `isLighthouseEnabled`, but that's not critical.

**Pull request merged**